crates/ruff/src/rules/flake8_unused_arguments/rules/unused_arguments.rs

```diff
@@ -436,23 +436,22 @@ pub(crate) fn unused_arguments(
                 }
             }
         }
-        ScopeKind::Lambda(ast::ExprLambda {
-            parameters: Some(parameters),
```

Comment:
Using `parameters: Some(parameters)` meant that we went to the panic branch if the lambda no parameters...